### PR TITLE
Add display registration and validation

### PR DESCRIPTION
**Pull request by tejohnso**
@ahmedalsudani @fjvallarino please review

### CODE REVIEW

File: installer/display-registration.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    return network.httpFetch(`${validationUrlBase}/display?id=${id}`)
+    .then((resp)=>{
+      return resp.json();
+    }).then((json)=>{
+      if (json.error) {return Promise.reject(json.error);}
+      return json;
+    });
+  },
+  submitClaimId(id, name) {
```
Comment (ahmedalsudani) [inline installer/display-registration.js]: @tejohnso why does the method above say `validate` and this one `submit?`

File: installer/display-registration.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if (!id) {return Promise.reject(new Error("no id provided"));}
+    if (!name) {return Promise.reject(new Error("no name provided"));}
+
+    return network.httpFetch(`${registrationUrlBase}/${id}/register?name=${name}`)
+    .then((resp)=>{
+      return resp.json();
+    }).then((json)=>{
+      if (json.status.code !== 0) {return Promise.reject(json.status.message);}
```
Comment (ahmedalsudani) [inline installer/display-registration.js]: @tejohnso why don't we check for `json.error` like we do above?

Comment (tejohnso) [inline installer/display-registration.js]: We have to check for what the api returns and the api's aren't consistent.  In one case it returns an error object, in the other it does not.

**Failed registration:**
```json
{ status: { code: 5, message: 'Company with this Claim ID is not found.' },
  displayId: '' }
```

**Failed validation:**
```json
{ error:
   { errors: [ [Object] ],
     code: 404,
     message: 'Display not found.' } }
```

File: installer/display-registration.js (hunk 1 above)
Comment (tejohnso) [inline installer/display-registration.js]: @ahmedalsudani In one case it's validating an existing display id.  In the other it's submitting a claim id to retrieve a new display id.

**Comment by ahmedalsudani** [comment]
:+1:

**Comment by fjvallarino** [comment]
:+1:

**Pull request merged**